File: lesson-1/login_script.py
from playwright.sync_api import sync_playwright
import requests

def solve_captcha(question):
        print(f'Solving captcha: {question}')
        # Call Ollama API
        response = requests.post('http://localhost:11434/api/chat',
            json={
                "model": "llama3.2:latest",
                "stream": False,
                "messages": [
                    {
                        "role": "user",
                        "content": f'{question} - answer as short as you can, only year, if able'
                    }
                ]
            }
        )
        # print(f'Ollama status code: {response.status_code}')
        data = response.json()
        print(f'Ollama response: {data['message']['content']}')
        return data['message']['content']

def login_with_captcha():
    try:
        with sync_playwright() as p:
            browser = p.chromium.launch(
                headless=True  # set to True in production
            )
            page = browser.new_page()
            page.goto('https://xyz.ag3nts.org/')

            # Wait for the captcha question to load
            captcha_question = page.locator('#human-question').text_content().strip()
            
            # Remove "Question:" prefix if present
            captcha_question = captcha_question.replace('Question:', '').strip()

            # Get answer from Ollama
            answer = solve_captcha(captcha_question)
            
            if not answer:
                raise Exception('Failed to get answer from Ollama')

            # Fill in the login form
            page.fill('input[name="username"]', 'tester')
            page.fill('input[name="password"]', '574e112a')
            page.fill('input[name="answer"]', answer)

            # Submit
            browser.close()
    except Exception as error:
        print('Error logging in:', error)

if __name__ == '__main__':
    login_with_captcha() 

File: lesson-1/login_script.js
const puppeteer = require('puppeteer');
const fetch = require('node-fetch');

async function solveCaptcha(question) {
    try {
        // Call Ollama API
        const response = await fetch('http://localhost:11434/api/generate', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                model: 'mistral',  // or another model you have in Ollama
                prompt: `Answer this question briefly with just the number or date: ${question}`
            })
        });

        const data = await response.json();
        // Extract the answer from Ollama's response
        return data.response.trim();
    } catch (error) {
        console.error('Error calling Ollama:', error);
        return null;
    }
}

async function loginWithCaptcha() {
    try {
        const browser = await puppeteer.launch({
            headless: false,  // set to true in production
            defaultViewport: null
        });

        const page = await browser.newPage();
        await page.goto('https://xyz.ag3nts.org/');

        // Wait for the captcha question to load
        const captchaQuestion = await page.$eval('.captcha-question', 
            el => el.textContent.trim()
        );

        // Get answer from Ollama
        const answer = await solveCaptcha(captchaQuestion);
        
        if (!answer) {
            throw new Error('Failed to get answer from Ollama');
        }

        // Fill in the login form
        await page.type('input[name="username"]', 'your_username');
        await page.type('input[name="password"]', 'your_password');
        await page.type('input[name="captcha"]', answer);

        // Submit
    } catch (error) {
        console.error('Error logging in:', error);
    } finally {
        await browser.close();
    }
}

loginWithCaptcha(); 